awx/main/managers.py

```diff
@@ -127,11 +129,21 @@ def register(self, uuid=None, hostname=None, ip_address=None, node_type='hybrid'
                         other_inst.save(update_fields=['ip_address'])
                         logger.warning("IP address {0} conflict detected, ip address unset for host {1}.".format(ip_address, other_hostname))
 
-            # Return existing instance that matches hostname
+            # get the instance based on the hostname
             instance = self.filter(hostname=hostname)
+
+            # Check or update the existing uuid
+            if uuid is not None and uuid != UUID_DEFAULT:
+                if self.filter(uuid=uuid).exists():
+                    instance = self.filter(uuid=uuid)
+
+            # Return existing instance
             if instance.exists():
                 instance = instance.get()
                 update_fields = []
+                if instance.hostname != hostname:
```

Comment:
Would add something like this to call out what's happening here:
`# if instance was retrieved by uuid and hostname has changed, update hostname`

Reply:
Would add a `logger.warn` here so that the logs reflect the hostname change happened.

Reply:
added line 145 should address that @jladdjr :)